demos/common/hfsm/az_hfsm.h

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file az_hfsm.h
+ * @brief Hierarchical Finite State Machine implementation.
+ */
+
+#ifndef _az_HFSM_H
+#define _az_HFSM_H
+
+#include <stdint.h>
+
+/**
+ * @brief hfsm event type.
+ *
+ */
+typedef enum
+{
+  AZ_HFSM_ENTRY = 1,
+  AZ_HFSM_EXIT = 2,
+  AZ_HFSM_ERROR = 3,
+  AZ_HFSM_TIMEOUT = 4,
+  AZ_HFSM_EVENT_BASE = 10,
```

Comment:
humm,,, really part of the enum?

Reply:
I'm thinking this should be more like https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/inc/azure/core/az_log.h#L44. 
That way each HFSM can have a different class and associated events can be identified during debugging. A common class would contain the ones here.